[Loh Joo Hoe] iP

lohjooh opened the pull request.

lohjooh committed:
Level-0

lohjooh committed:
Revert "Level-0"

This reverts commit a48260df3d6e74ea4c8a1a6df41a356fe6749910.

lohjooh committed:
-Added greeting message
-Added shutdown message

lohjooh committed:
-Changed name of file to be the same as bot name
-Added echo function
-Added shutdown function

lohjooh committed:
-Added function to remember commands (up to 100 commands)
-Added function for "list" command

lohjooh committed:
-Removed unused Duke.java file
-Added Task class (Task.java)
-Added function for "mark"
-Added function for "unmark"
-Changed "list" to use Task class
-Changed ASCII art to static final variables
-Changed "add" function to use Task class

lohjooh committed:
-Fixed formatting for ASCII art

lohjooh committed:
-Fixed not taking in mulitple words

lohjooh committed:
-Fixed program crashing when using mark command with a string

lohjooh committed:
-Changed if-else statements to switch-case statements

lohjooh committed:
Added: Child classes of "Task" for Level-4

Added: Support for "Deadline", "Event", "ToDo"

Improved: Flavour text

Fixed: Bugs involving "mark" and "unmark" cases

Fixed: Constants not having uppercase names

Added: Divider to each reply from the bot

Added: There is now a toString overload function in "Task"

Added: Type is now tracked in "Task"

Changed: Default case no longer adds a task, instead it prompts the user again

Changed: "Task" no longer used as a standalone type

Fixed: Error message no longer print out multiple times when an invalid input is typed

lohjooh committed:
Changed: Name of .java file back to Duke to prevent script errors

Tried: Ran runtest.bat and changed EXPECTED to expected output and added input test case

lohjooh committed:
Added: DukeException for error handing

Changed: Made ASCII art smaller

Changed: Existing error checking now uses exceptions

lohjooh committed:
Fixed: Removed bugs caused by new error handing

lohjooh committed:
Changed: ASCII goodbye art;

lohjooh committed:
Merge branch 'master' into branch-Level-5

lohjooh committed:
Fixed: Made some class variables protected

saintzaw commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,170 @@
+        return num;
+    }
+
+    public static void checkSameDone(int index, boolean changeTo, String type) throws DukeException{
```
Perhaps you can consider changing the method name to something more descriptive, I am not very clear on what this is for.

saintzaw commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,170 @@
+            }
+            System.out.println(DIV);
+            switch (checkCmd) {
+            case "list": {
```
I think the brackets after case "list" : and other cases are not necessary, removing the brackets might make it look neater

saintzaw commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,170 @@
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        hello();
```
Your main method looks neat with the same level of abstraction, it's satisfying to see.

saintzaw commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,170 @@
+            "██████╦╝╚█████╔╝██████╦╝\n" +
+            "╚═════╝░░╚════╝░╚═════╝░";
+    static private final String BYE = "\n" +
+            "██████╗░░█████╗░██████╗░  ░██████╗░█████╗░██╗░░░██╗░██████╗  ██████╗░██╗░░░██╗███████╗\n" +
```
I like this, really creative way of saying goodbye!

saintzaw reviewed (commented):
I think you've done a great job so far! Keep it up :)

lohjooh committed:
Fixed: Rectified bugs introduced in recent changes

lohjooh committed:
Fixed: Removed some extraneous braces

lohjooh committed:
Added: Grouped all classes into packages

Added: Added echo method to improve readability

lohjooh committed:
Replaced: Changed fixed size list to ArrayLists

Added: Added skeleton code for help function

lohjooh committed:
Fixed: Changed implementation of list number such that it works in line with the deletion

Added: Function to delete list items based on the list number

lohjooh committed:
Added: Read from file and write from file

Fixed: Improved code quality

lohjooh committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Added: Read from file and write from file

# Conflicts:
#	src/main/java/Duke.java

lohjooh committed:
Fixed: Resolved merge conflict

Fixed: Corrected bugs caused by saving file

lohjooh committed:
Added: Function to create directory for data text file and text file itself added.

Fixed: Resolved bug causing reading of file to fail

lohjooh committed:
Fixed: Resolved issues with event getTask() member methods

Fixed: Resolved issues with writing event objects into data text file

lohjooh committed:
Added: FileManager, DataManager, DukeMessages classes for better management

DukeMessages: Error message and add messages stored within along with printing functionality

DataManager: Separately stored data in this class, methods to modify data moved to this class as well

FileManager: Used for reading and writing .txt file stored locally

lohjooh committed:
Added: Improved implementation of reading files such that the print messaged is not displayed

Changed: Changed write file implementation such that the file is changed immediately after changes to checklist have been made

lohjooh committed:
Added: Jar file

lohjooh committed:
Added: More classes such that the project is more modular

lohjooh committed:
Added: Functionality to recognise dates in a specific format, minimal date functionality achieved

mathanmahe commented on src/main/duke/Duke.java:
```diff
@@ -0,0 +1,68 @@
+    private final DataManager dm;
+    private final DukeMessages ui;
+    private final Parser parser;
+    //    static private final HashMap<String, String> helpOutputs = new HashMap<>();
```
Consider removing dead code, to improve readability. Not to worry if you are not aware of this coding quality violation, as it will be covered later in the lectures.

mathanmahe commented on src/main/duke/Duke.java:
```diff
@@ -0,0 +1,68 @@
+    public void run() {
+        dm.run();
+        do {
+            String checkCmd = parser.run();
```
Code quality guideline: Name well. 
Try to use nouns for things and verbs for actions. Here instead of using check verb in `checkCmd`, which sounds more like a function, you can consider renaming to `parsedCommand`.

mathanmahe commented on src/main/duke/storage/FileManager.java:
```diff
@@ -0,0 +1,97 @@
+    private static final ArrayList<String> storedData = new ArrayList<>();
+
+    public static void readFile(String path) throws FileNotFoundException {
+        File f = new File(path);
```
Code Quality guideline: Name well
Consider giving more descriptive names like `newfile` or `scannerObject`, to improve readability.

mathanmahe commented on src/main/duke/tasklist/DataManager.java:
```diff
@@ -0,0 +1,243 @@
+//                String next = in.nextLine();
+//                printHelp(next.trim());
+//                break;
+        case "mark":
```
Can consider having the switch case inside try catch, as you seem to be catching the same exception in each case. This would improve readability as well.

mathanmahe commented on src/main/duke/tasklist/DataManager.java:
```diff
@@ -0,0 +1,243 @@
+    }
+
+    public void run() {
+        ui.printDiv();
```
Good use of SLAP

mathanmahe commented on src/main/duke/tasklist/DataManager.java:
```diff
@@ -0,0 +1,243 @@
+import duke.storage.FileManager;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
```
Java Coding Standard: The ordering of import statements must be consistent.
Group all the `utils` together and the `io` together, and so on, for example.

mathanmahe reviewed (commented):
Good general adherence to most coding standards, spotted some minor areas for improvement in naming, readability and coding standard.

lohjooh committed:
Fixed: Improved readability of code based on suggestions

Added: More modularity for tasklist object

lohjooh committed:
Added: Command "date" to search list by date

Changed: Implementation of list changed to hashmap

lohjooh committed:
Added: Skeleton code for find function

lohjooh committed:
Added: Prepared parser class for find function

lohjooh committed:
Merge pull request #1 from lohjooh/branch-Level-8

Branch level 8

lohjooh committed:
Merge branch 'master' of https://github.com/lohjooh/ip

* 'master' of https://github.com/lohjooh/ip:
  Added: Command "date" to search list by date
  Fixed: Improved readability of code based on suggestions
  Added: Functionality to recognise dates in a specific format, minimal date functionality achieved

lohjooh committed:
Merge branch 'master' into branch-Level-9

lohjooh committed:
Merge pull request #2 from lohjooh/branch-Level-9

Branch level 9

lohjooh committed:
Merge branch 'master' of https://github.com/lohjooh/ip

* 'master' of https://github.com/lohjooh/ip:
  Added: Prepared parser class for find function
  Added: Skeleton code for find function

lohjooh committed:
Fixed: Resolved bugs when using "next" keyword with "date"

Added: Custom message for "find" function

lohjooh committed:
Added: Full functionality for "find"

Fixed: Resolved issue with ToDo not working with "find"

lohjooh committed:
Merge branch 'branch-Level-9' of https://github.com/lohjooh/ip into branch-Level-9

* 'branch-Level-9' of https://github.com/lohjooh/ip:
  Added: Command "date" to search list by date
  Fixed: Improved readability of code based on suggestions
  Added: Functionality to recognise dates in a specific format, minimal date functionality achieved

lohjooh committed:
Merge branch 'branch-Level-9'

lohjooh committed:
Added: Javadoc documentation

lohjooh committed:
Added: Clear to clear all data

Removed: Deleted function for deleting datafile as clear function makes it obsolete

Updated: Text UI messages update

Added: User guide made

lohjooh committed:
Fixed: Bug where fromDate is not stored properly

lohjooh committed:
Fixed: Fixed bug with date formatting user input

lohjooh committed:
Added: A warning section for blank fields when adding tasks

lohjooh committed:
Fixed: Fixed bug when deleting items

lohjooh committed:
Fixed: Ignores unnecessary input for 'list' and 'clear'

lohjooh committed:
Added: Warning about whitespaces

lohjooh committed:
Added: Reverse changes to retroactively add branch

lohjooh committed:
Merge branch 'branch-A-JavaDoc'

lohjooh committed:
Merge branch 'master' of https://github.com/lohjooh/ip

lohjooh committed:
Update README.md
